treeherder/etl/buildapi.py

```diff
@@ -62,32 +62,31 @@ def find_job_guid(self, build):
                     prop["branch"], build))
                 raise e
 
-        request_ids_str = ",".join(map(str, request_ids))
-        request_time_list = []
-
-        if type(request_times) == dict:
-            for request_id in request_ids:
-                request_time_list.append(
-                    request_times[str(request_id)])
-            request_times_str = ','.join(
-                map(str, request_time_list))
-        else:
-            request_times_str = str(request_times)
+        request_id = request_ids[-1]
+        request_time = request_times[str(request_id)]
 
         job_guid_data = {'job_guid': '', 'coalesced': []}
 
         if len(request_ids) > 1:
-            # coallesced job detected, generate the coalesced
-            # job guids
-            for index, r_id in enumerate(request_ids):
-                #skip if buildbot doesn't have a matching number of ids and times
-                if len(request_time_list) > index:
+            # coallesced job detected, generate the coalesced job guids
+
+            # delete the last request_id, which is the current job, because
+            # it was not, itself, coalesced.  The other requests were
+            # coalesced to it.
+            request_ids.pop()
+
+            # build the list of job_guids that were coalesced
+            for r_id in request_ids:
+                r_id = str(r_id)
```

Comment:
I don't think the str() is necessary, even though the old code used it. generate_job_guid does a str() itself:
https://github.com/mozilla/treeherder-service/blob/40b01c751d4e2500cb069933ac646ea32249eafc/treeherder/etl/common.py#L128

There are probably other places in the file where it could be removed :-)